src/Bicep.Core/Syntax/ArrayAccessSyntax.cs

```diff
@@ -4,7 +4,7 @@
 
 namespace Bicep.Core.Syntax
 {
-    public class ArrayAccessSyntax : SyntaxBase
+    public class ArrayAccessSyntax : ExpressionSyntax
```

Comment:
Do you think there's ever going to be a need to attach some other kind of 'X is a Y' information which can overlap with being an expression? I was wondering about e.g. a flags-based approach, but this makes sense if multiple inheritance is not needed.

Reply:
Possibly? I think we have some `kind` enums in a few places and flags in others. I've also been using interfaces as an attribute of sorts in a few places because they play nicely with the `is` operator and pattern matching in c#.